Client/Assets/Passwords.js

```diff
@@ -0,0 +1,699 @@
+      //} else {
+        //didntFillReason = "multipleLogins";
+        //log("Multiple logins for form, so not filling any.");
+      //}
+    }
+
+    var didFillForm = false;
+    if (selectedLogin && autofillForm && !isFormDisabled) {
+      // Fill the form
+      if (usernameField) {
+        // Don't modify the username field if it's disabled or readOnly so we preserve its case.
+        var disabledOrReadOnly = usernameField.disabled || usernameField.readOnly;
+
+        var userNameDiffers = selectedLogin.username != usernameField.value;
+        // Don't replace the username if it differs only in case, and the user triggered
+        // this autocomplete. We assume that if it was user-triggered the entered text
+        // is desired.
+        var userEnteredDifferentCase = userTriggered && userNameDiffers && usernameField.value.toLowerCase() == selectedLogin.username.toLowerCase();
+
+        if (!disabledOrReadOnly && !userEnteredDifferentCase && userNameDiffers) {
+          // usernameField.setUserInput(selectedLogin.username);
+          usernameField.value = selectedLogin.username
+        }
+      }
+      if (passwordField.value != selectedLogin.password) {
+        // passwordField.setUserInput(selectedLogin.password);
+        passwordField.value = selectedLogin.password
+      }
+      didFillForm = true;
+    } else if (selectedLogin && !autofillForm) {
+      // For when autofillForm is false, but we still have the information
+      // to fill a form, we notify observers.
+      didntFillReason = "noAutofillForms";
+      // Services.obs.notifyObservers(form, "passwordmgr-found-form", didntFillReason);
+      log("autofillForms=false but form can be filled; notified observers");
+    } else if (selectedLogin && isFormDisabled) {
+      // For when autocomplete is off, but we still have the information
+      // to fill a form, we notify observers.
+      didntFillReason = "autocompleteOff";
+      // Services.obs.notifyObservers(form, "passwordmgr-found-form", didntFillReason);
+      log("autocomplete=off but form can be filled; notified observers");
+    }
+
+    // this._notifyFoundLogins(didntFillReason, usernameField, passwordField, foundLogins, selectedLogin);
+    return [didFillForm, foundLogins];
+  },
+}
+
+var LoginUtils = {
+  /*
+   * _getPasswordOrigin
+   *
+   * Get the parts of the URL we want for identification.
+   */
+  _getPasswordOrigin : function (uriString, allowJS) {
+    // All of this logic is moved to swift (so that we don't need a uri parser here)
+    return uriString;
+  },
+
+  _getActionOrigin : function(form) {
+    var uriString = form.action;
+
+    // A blank or missing action submits to where it came from.
+    if (uriString == "")
+      uriString = form.baseURI; // ala bug 297761
+
+    return this._getPasswordOrigin(uriString, true);
+  },
+}
+
+function onBlur(event) {
+  LoginManagerContent.onUsernameInput(event)
+}
+
+window.addEventListener("load", function(event) {
+  try {
```

Comment:
Maybe we could start doing something like <a href="http://pastebin.mozilla.org/8702302">this</a> for all functions defined in content scripts?

Don't have to do this here, just thinking about ways to make this better.